overlord/configstate/configcore/corecfg.go

```diff
@@ -57,6 +58,7 @@ var supportedConfigurations = map[string]bool{
 	"core.experimental.layouts":            true,
 	"core.experimental.parallel-instances": true,
 	"core.experimental.hotplug":            true,
+	"core.experimental.snapd-snap":         true,
```

Comment:
+1 but just thinking about this it looks like a place where we could hold validation function instead of the true value here. Then we can simply look up the validation function below.